unstructured/embed/bedrock.py

```diff
@@ -0,0 +1,72 @@
+import types
+from typing import List
+
+import numpy as np
+import boto3
+
+from unstructured.documents.elements import (
+    Element,
+)
+from unstructured.embed.interfaces import BaseEmbeddingEncoder
+from unstructured.ingest.error import EmbeddingEncoderConnectionError
+from unstructured.utils import requires_dependencies
+from langchain.embeddings import BedrockEmbeddings
+
+class BedrockEmbeddingEncoder(BaseEmbeddingEncoder):
+    def __init__(self, aws_access_key_id: str, aws_secret_access_key: str, region_name: str = "us-west-2"):
+        self.aws_access_key_id = aws_access_key_id
+        self.aws_secret_access_key = aws_secret_access_key
+        self.region_name = region_name
+        self.initialize()
+
+    def initialize(self):
+        self.bedrock_client = self.get_bedrock_client()
+
+    def num_of_dimensions(self):
+        return np.shape(self.examplary_embedding)
+
+    def is_unit_vector(self):
+        return np.isclose(np.linalg.norm(self.examplary_embedding), 1.0)
+
+    def embed_query(self, query):
+        return np.array(self.bedrock_client.embed_query(query))
+
+    def embed_documents(self, elements: List[Element]) -> List[Element]:
+        embeddings = [np.array(self.bedrock_client.embed_query(str(e))) for e in elements]
+        elements_with_embeddings = self._add_embeddings_to_elements(elements, embeddings)
+        return elements_with_embeddings
+
+    def _add_embeddings_to_elements(self, elements, embeddings) -> List[Element]:
+        assert len(elements) == len(embeddings)
+        elements_w_embedding = []
+
+        for i, element in enumerate(elements):
+            original_method = element.to_dict
+
+            def new_to_dict(self):
+                d = original_method()
+                d["embeddings"] = self.embeddings
+                return d
+
+            element.embeddings = embeddings[i]
+            elements_w_embedding.append(element)
+            element.to_dict = types.MethodType(new_to_dict, element)
+        return elements
```

Comment:
I see this is using https://github.com/Unstructured-IO/unstructured/blob/97f832619ab60022b29c50ba89703d9922cfbe5f/unstructured/embed/openai.py#L37-L52
this part of the code essentially depend on `element` data definition so can be shared by different embeddings. So we can consider refactor this either:
- as a method for elements
- as a utils func in `unstructured/embed/utils.py` so both here and openai version can import the same function